### PR TITLE
Create student repos

### PR DESCRIPTION
**Pull request by ArmaanT**
This PR adds terraform configuration to create student homework repositories

### CODE REVIEW

File: terraform/main.tf [hunk 1]
```diff
@@ -1,11 +1,20 @@
...
+    hw2           = false
+    hw3           = false
+    hw4           = false
+    final_project = false
+  }
+  // Create a list of objects representing all homework repos to be (eventually) created. 
+  hws = flatten([
+    for hw, published in local.published : [
+      for student in var.students : {
+        hw = hw, student = student, published = published
```
Comment (pawalt) [inline terraform/main.tf:15]: this shit is SO cursed

File: terraform/modules/hw_repo/main.tf [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+  visibility  = "private"
+
+  template {
+    owner = "cis188"
+    repository = var.hw
+  }
+}
+
+resource "github_team_repository" "hw" {
+  count = var.published ? 1 : 0
```
Comment (pawalt) [inline terraform/modules/hw_repo/main.tf:13]: > us: hi can we have if statements in tf?
> hashicorp: no we have if statements at home
> if statements at home:

**Review by pawalt — APPROVED** [review]
nicejobmaan

File: terraform/main.tf (hunk 1 above)
Comment (ArmaanT) [reply terraform/main.tf:15]: My favorite part about this is that it's officially recommended

**Pull request merged**